godot-core/src/builtin/script.rs

```diff
@@ -0,0 +1,818 @@
+        has_method_func: Some(script_instance_info::has_method_func::<T>),
+        call_func: Some(script_instance_info::call_func::<T>),
+        // deprecated by Godot.
+        notification_func: None,
+        to_string_func: Some(script_instance_info::to_string_func::<T>),
+        refcount_decremented_func: Some(script_instance_info::refcount_decremented_func::<T>),
+        refcount_incremented_func: Some(script_instance_info::refcount_incremented_func::<T>),
+        is_placeholder_func: Some(script_instance_info::is_placeholder_func::<T>),
+        // usage is unclear.
+        set_fallback_func: None,
+    };
+
+    let instance_ptr = Box::into_raw(Box::new(gd_instance));
+
+    let data = ScriptInstanceData {
+        inner: RefCell::new(rs_instance),
+        gd_instance_ptr: instance_ptr,
+    };
+
+    let data_ptr = Box::into_raw(Box::new(data));
+
+    // SAFETY: `script_instance_create` expects a `GDExtensionScriptInstanceInfoPtr` and a generic `GDExtensionScriptInstanceDataPtr` of our
+    // choice. The validity of the instance info struct is ensured by code generation.
+    unsafe {
+        sys::interface_fn!(script_instance_create)(
+            instance_ptr,
+            data_ptr as sys::GDExtensionScriptInstanceDataPtr,
+        )
+    }
+}
+
+mod script_instance_info {
+    use std::any::type_name;
+    use std::cell::{BorrowError, Ref, RefMut};
+    use std::ffi::c_void;
+    use std::mem::ManuallyDrop;
+
+    use crate::builtin::meta::GodotType;
+    use crate::builtin::{GString, StringName, Variant};
+    use crate::engine::{Object, ScriptLanguage};
+    use crate::obj::Gd;
+    use crate::private::handle_panic;
+    use crate::sys::{self, GodotFfi};
+
+    use super::{ScriptInstance, ScriptInstanceData};
+
+    fn borrow_instance_mut<'a, T: ScriptInstance>(
+        instance: *mut ScriptInstanceData<T>,
+    ) -> RefMut<'a, T> {
+        unsafe { (*instance).inner.borrow_mut() }
+    }
+
+    fn borrow_instance<'a, T: ScriptInstance>(instance: *mut ScriptInstanceData<T>) -> Ref<'a, T> {
+        unsafe { (*instance).inner.borrow() }
+    }
+
+    fn try_borrow_instance<'a, T: ScriptInstance>(
+        instance: *mut ScriptInstanceData<T>,
+    ) -> Result<Ref<'a, T>, BorrowError> {
+        unsafe { (*instance).inner.try_borrow() }
+    }
+
+    /// # Safety
+    ///
+    /// - We expect the engine to provide a vallid instance pointer.
+    fn instance_data_as_script_instance<T: ScriptInstance>(
+        p_instance: sys::GDExtensionScriptInstanceDataPtr,
+    ) -> *mut ScriptInstanceData<T> {
+        p_instance as *mut ScriptInstanceData<T>
+    }
+
+    /// # Safety
+    ///
+    /// - We expect the engine to provide a valid const string name pointer.
+    unsafe fn transfer_string_name_from_godot(
+        p_name: sys::GDExtensionConstStringNamePtr,
+    ) -> StringName {
+        StringName::from_arg_ptr(
+            p_name as sys::GDExtensionTypePtr,
+            sys::PtrcallType::Standard,
+        )
+    }
```

Comment:
Isn't this just `StringName::from_string_sys(sys::force_mut_ptr(p_name))`?

Reply:
We can not run the `StringName` destructor for this. It is a case like this: https://github.com/godot-rust/gdext/blob/0b164abb6371d621d8fa1fbba6b54a008c5ea112/godot-core/src/registry.rs#L581-L584

Reply:
applied the suggested change.